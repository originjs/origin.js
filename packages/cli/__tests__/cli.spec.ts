import { createServer } from 'vite'
import run, { DEMO_PATH } from '../../cli-test-utils/execCommands'
import create from '../../cli-test-utils/createTestProject'
import runServer from '../../cli-test-utils/createTestProjectServer'
import runBuild from '../../cli-test-utils/buildTestProject'

test('ori -h', () => {
  const { stdout, exitCode } = run(['-h'])
  expect(stdout).toMatchSnapshot('A1')
  expect(exitCode).toBe(0)
})

test('ori --help', () => {
  const { stdout, exitCode } = run(['--help'])
  expect(stdout).toMatchSnapshot('A2')
  expect(exitCode).toBe(0)
})

test('ori init without app-name', async () => {
  try {
    await create('')
  } catch (e) {
    expect(e.stderr).toContain('Failed to initialize the template')
    expect(e.stderr).toContain("error: missing required argument 'app-name'")
  }
})

test('ori init with failed arguments', async () => {
  try {
    await create('', ['-a'])
  } catch (e) {
    expect(e.stderr).toContain('Failed to initialize the template')
  }
})

test('ori init with all plugins', async () => {
  const project = await create('testAllPlugins', ['-d', '-a'])

  expect(project.has('index.html')).toBe(true)
  const indexContent = await project.read('index.html')
  expect(indexContent).toMatch(
    `<script type="module" src="/src/main.ts"></script>`,
  )

  expect(project.has('./src/main.ts')).toBe(true)
  expect(project.has('./src/App.vue')).toBe(true)
  expect(project.has('./src/pages')).toBe(true)
  expect(project.has('./src/layouts')).toBe(true)
  expect(project.has('./src/components')).toBe(true)
  expect(project.has('./src/assets')).toBe(true)

  expect(project.has('package.json')).toBe(true)
  const packageConfig = await project.read('package.json')
  expect(packageConfig).toMatchSnapshot('A3')

  expect(project.has('vite.config.ts')).toBe(true)
  const viteConfig = await project.read('vite.config.ts')
  expect(viteConfig).toMatchSnapshot('A3')

  project.clear()
}, 15000)

test('ori init without plugins', async () => {
  const project = await create('testNoPlugins')

  expect(project.has('index.html')).toBe(true)
  const indexContent = await project.read('index.html')
  expect(indexContent).toMatch(
    `<script type="module" src="/src/main.ts"></script>`,
  )

  expect(project.has('./src/main.ts')).toBe(true)
  expect(project.has('./src/App.vue')).toBe(true)

  expect(project.has('package.json')).toBe(true)
  const packageConfig = await project.read('package.json')
  expect(packageConfig).toMatchSnapshot('A4')

  expect(project.has('vite.config.ts')).toBe(true)
  const viteConfig = await project.read('vite.config.ts')
  expect(viteConfig).toMatchSnapshot('A4')

  project.clear()
}, 10000)

test('ori init --help', () => {
  const { stdout, exitCode } = run(['init', '--help'])
  expect(stdout).toMatchSnapshot('A5')
  expect(exitCode).toBe(0)
})

test.skip('ori dev', async () => {
  const project = await create('testServer')

  await runServer('testServer')
  expect(createServer).toHaveBeenCalledTimes(1)
  // TODO: write files and update changes

  project.clear()
}, 10000)

test.skip('ori build', async () => {
  const project = await create('testBuild')

  await runBuild('testBuild')
  expect(project.has('dist')).toBe(true)

  project.clear()
}, 10000)

test.skip('ori tovue3', () => {
  const { stdout, exitCode } = run(['tovue3', 'src', '-a'])
  expect(stdout).toMatchSnapshot('A7')
  expect(exitCode).toBe(0)
})

test.skip('ori tovite', () => {
  const { stdout, exitCode } = run(['tovite', '-d', DEMO_PATH])
  expect(stdout).toContain('Conversion finished')
  expect(exitCode).toBe(0)
})
