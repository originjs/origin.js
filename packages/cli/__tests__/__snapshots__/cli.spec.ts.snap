// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ori --help: A2 1`] = `
"Usage: ori <command> [options]

Options:
  -h, --help                 display help for command

Commands:
  init [options] <app-name>  init a new project
  dev                        alias of \\"ori dev\\" in the current project
  build                      alias of \\"ori build\\" in the current project
  tovite [options]           use vite in the current project
  tovue3 [options]           use vue-next in the current project
  help [command]             display help for command"
`;

exports[`ori -h: A1 1`] = `
"Usage: ori <command> [options]

Options:
  -h, --help                 display help for command

Commands:
  init [options] <app-name>  init a new project
  dev                        alias of \\"ori dev\\" in the current project
  build                      alias of \\"ori build\\" in the current project
  tovite [options]           use vite in the current project
  tovue3 [options]           use vue-next in the current project
  help [command]             display help for command"
`;

exports[`ori init --help: A5 1`] = `
"Usage: ori init [options] <app-name>

init a new project

Options:
  -d,--default      skip init project options
  -a,--all-plugins  create project with all plugins
  -h, --help        display help for command"
`;

exports[`ori init with all plugins: A3 1`] = `
"{
  \\"name\\": \\"testAllPlugins\\",
  \\"version\\": \\"1.0.0\\",
  \\"license\\": \\"ISC\\",
  \\"scripts\\": {
    \\"dev\\": \\"vite\\",
    \\"build\\": \\"vue-tsc --noEmit && vite build\\",
    \\"serve\\": \\"vite preview\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^3.0.5\\",
    \\"vue-router\\": \\"^4.0.11\\"
  },
  \\"devDependencies\\": {
    \\"@vitejs/plugin-vue\\": \\"^1.3.0\\",
    \\"@vue/compiler-sfc\\": \\"^3.0.5\\",
    \\"@originjs/vite-plugin-assets\\":\\"latest\\",
    \\"unplugin-vue-components\\":\\"latest\\",
    \\"@originjs/vite-plugin-content\\":\\"latest\\",
    \\"@originjs/vite-plugin-pages\\":\\"latest\\",
    \\"typescript\\": \\"^4.3.2\\",
    \\"vite\\": \\"^2.4.4\\",
    \\"vue-tsc\\": \\"^0.2.2\\"
  }
}
"
`;

exports[`ori init with all plugins: A3 2`] = `
"import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import assets from '@originjs/vite-plugin-assets'
import Components from 'unplugin-vue-components'
import content from '@originjs/vite-plugin-content'
import pages from '@originjs/vite-plugin-pages'

// https://vitejs.dev/config/
export default defineConfig({
  plugins:[
    vue(),
    assets(),
    Components({
      // allow auto load markdown components under \`./src/components/\`
      extensions: ['vue', 'md'],
      // allow typescript
      dts: true,
      // allow auto import and register components used in markdown
      include: [/\\\\.vue$/, /\\\\.vue\\\\?vue/, /\\\\.md$/],
    }),
    content(),
    pages()
  ]
})
"
`;

exports[`ori init without plugins: A3 1`] = `
"{
  \\"name\\": \\"testAllPlugins\\",
  \\"version\\": \\"1.0.0\\",
  \\"license\\": \\"ISC\\",
  \\"scripts\\": {
    \\"dev\\": \\"vite\\",
    \\"build\\": \\"vue-tsc --noEmit && vite build\\",
    \\"serve\\": \\"vite preview\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^3.0.5\\",
    \\"vue-router\\": \\"^4.0.11\\"
  },
  \\"devDependencies\\": {
    \\"@vitejs/plugin-vue\\": \\"^1.3.0\\",
    \\"@vue/compiler-sfc\\": \\"^3.0.5\\",
    \\"@originjs/vite-plugin-assets\\":\\"latest\\",
    \\"unplugin-vue-components\\":\\"latest\\",
    \\"@originjs/vite-plugin-content\\":\\"latest\\",
    \\"@originjs/vite-plugin-pages\\":\\"latest\\",
    \\"typescript\\": \\"^4.3.2\\",
    \\"vite\\": \\"^2.4.4\\",
    \\"vue-tsc\\": \\"^0.2.2\\"
  }
}
"
`;

exports[`ori init without plugins: A3 2`] = `
"import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import assets from '@originjs/vite-plugin-assets'
import Components from 'unplugin-vue-components'
import content from '@originjs/vite-plugin-content'
import pages from '@originjs/vite-plugin-pages'

// https://vitejs.dev/config/
export default defineConfig({
  plugins:[
    vue(),
    assets(),
    Components({
      // allow auto load markdown components under \`./src/components/\`
      extensions: ['vue', 'md'],
      // allow typescript
      dts: true,
      // allow auto import and register components used in markdown
      include: [/\\\\.vue$/, /\\\\.vue\\\\?vue/, /\\\\.md$/],
    }),
    content(),
    pages()
  ]
})
"
`;

exports[`ori init without plugins: A4 1`] = `
"{
  \\"name\\": \\"testNoPlugins\\",
  \\"version\\": \\"1.0.0\\",
  \\"license\\": \\"ISC\\",
  \\"scripts\\": {
    \\"dev\\": \\"vite\\",
    \\"build\\": \\"vue-tsc --noEmit && vite build\\",
    \\"serve\\": \\"vite preview\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^3.0.5\\",
    \\"vue-router\\": \\"^4.0.11\\"
  },
  \\"devDependencies\\": {
    \\"@vitejs/plugin-vue\\": \\"^1.3.0\\",
    \\"@vue/compiler-sfc\\": \\"^3.0.5\\",
    \\"typescript\\": \\"^4.3.2\\",
    \\"vite\\": \\"^2.4.4\\",
    \\"vue-tsc\\": \\"^0.2.2\\"
  }
}
"
`;

exports[`ori init without plugins: A4 2`] = `
"import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'

// https://vitejs.dev/config/
export default defineConfig({
  plugins:[
    vue()
  ]
})
"
`;
